### PR TITLE
Fix schema vendor extensions

### PR DESCRIPTION
**Pull request by immoh**
The current implementation didn't allow extension properties at all and they were also not tested at all.

I believe it is not possible express definitions for Responses and Paths objects using Schema but I've added TODOs if somebody has an idea how to do it.

### CODE REVIEW

**Comment by harti2006** [comment]
@immoh Will you continue working on the TODOs?

I'm not sure if the current state will break existing projects.

**Comment by immoh** [comment]
@harti2006 This change should be backwards compatible: it relaxes schema validation by allowing vendor extensions in certain places. The previous implementation looked like it allowed vendor extensions but it was not tested and actually didn't allow them at all. 

TODOs are about places where I wasn't able to modify schema so that it would allow vendor extensions. It didn't allow them before this change either, so the change shouldn't break anything. I have no plans to work on them at the moment.

**Comment by harti2006** [comment]
👍

**Comment by dryewo** [comment]
👍

File: src/io/sarnowski/swagger1st/schemas/swagger_2_0.clj [hunk 1]
```diff
@@ -1,5 +1,6 @@
 (ns io.sarnowski.swagger1st.schemas.swagger-2-0
-  (:require [schema.core :as s]))
+  (:require [clojure.string]
```
Comment (dixel) [inline src/io/sarnowski/swagger1st/schemas/swagger_2_0.clj]: I've seen `:require [clojure.string :as string]` in this project more often, would be cool to have it consistent and I am also fan of more explicit things.

Comment (oporkka) [reply src/io/sarnowski/swagger1st/schemas/swagger_2_0.clj]: I agree. Let's change it to `[clojure.string :as string]`

Comment (immoh) [reply src/io/sarnowski/swagger1st/schemas/swagger_2_0.clj]: I have no idea what "more explicit things" means here but I've changed the require to be consistent.

**Comment by immoh** [comment]
Force pushed, please approve again.

**Comment by harti2006** [comment]
:+1:

**Comment by oporkka** [comment]
:+1:

**Pull request merged**